packages/next/build/swc/src/styled_jsx/transform_css.rs

```diff
@@ -0,0 +1,256 @@
+use easy_error::{bail, Error};
+use swc_common::{source_map::Pos, BytePos, Span, SyntaxContext, DUMMY_SP};
+use swc_css::ast::*;
+use swc_css::parser::{parse_str, parse_tokens, parser::ParserConfig};
+use swc_css::visit::{VisitMut, VisitMutWith};
+use swc_css_codegen::{
+  writer::basic::{BasicCssWriter, BasicCssWriterConfig},
+  CodegenConfig, Emit,
+};
+use swc_ecmascript::ast::{Expr, Str, StrKind, Tpl, TplElement};
+use swc_ecmascript::utils::HANDLER;
+use swc_stylis::prefixer::prefixer;
+
+use super::{hash_string, string_literal_expr, LocalStyle};
+
+pub fn transform_css(
+  style_info: &LocalStyle,
+  is_global: bool,
+  class_name: &Option<String>,
+) -> Result<Expr, Error> {
+  let result: Result<Stylesheet, _> = parse_str(
+    &style_info.css,
+    style_info.css_span.lo,
+    style_info.css_span.hi,
+    ParserConfig {
+      parse_values: false,
+    },
+  );
+  let mut ss = match result {
+    Ok(ss) => ss,
+    Err(_) => {
+      HANDLER.with(|handler| {
+        handler
+          .struct_span_err(
+            style_info.css_span,
+            "Failed to parse css in styled jsx component",
+          )
+          .emit()
+      });
+      bail!("Failed to parse css");
+    }
+  };
+  // ? Do we need to support optionally prefixing?
+  ss.visit_mut_with(&mut prefixer());
+  ss.visit_mut_with(&mut Namespacer {
+    class_name: match class_name {
+      Some(s) => s.clone(),
+      None => format!("jsx-{}", &hash_string(&style_info.hash)),
+    },
+    is_global,
+    is_dynamic: style_info.is_dynamic,
+  });
+
+  let mut s = String::new();
+  {
+    let mut wr = BasicCssWriter::new(&mut s, BasicCssWriterConfig { indent: "  " });
+    let mut gen = swc_css_codegen::CodeGenerator::new(&mut wr, CodegenConfig { minify: true });
+
+    gen.emit(&ss).unwrap();
+  }
+
+  if style_info.expressions.len() == 0 {
+    return Ok(string_literal_expr(&s));
+  }
+
+  let mut parts: Vec<&str> = s.split("__styled-jsx-placeholder__").collect();
+  let mut final_expressions = vec![];
+  for i in 1..parts.len() {
+    let expression_index = parts[i].chars().nth(0).unwrap().to_digit(10).unwrap() as usize;
+    final_expressions.push(style_info.expressions[expression_index].clone());
+    let substr = &parts[i][1..];
+    parts[i] = substr;
+  }
+
+  Ok(Expr::Tpl(Tpl {
+    quasis: parts
+      .iter()
+      .map(|quasi| TplElement {
+        cooked: None, // ? Do we need cooked as well
+        raw: Str {
+          value: String::from(*quasi).into(),
```

Comment:
You can create `JsWord` from `&str`. If type is not &str, but a thing which implements `Deref<Target = str>`, you can do `(*quasi).into()`.
(Possibly `(**quasi).into()`)